src/qt/paymentrequestplus.cpp

```diff
@@ -150,7 +152,13 @@ bool PaymentRequestPlus::getMerchant(X509_STORE* certStore, QString& merchant) c
         int result = X509_verify_cert(store_ctx);
         if (result != 1) {
             int error = X509_STORE_CTX_get_error(store_ctx);
-            throw SSLVerifyError(X509_verify_cert_error_string(error));
+            // For testing payment requests, we allow self signed root certs!
+            // This option is just shown in the UI options, if -help-debug is enabled.
+            if (!(error == X509_V_ERR_DEPTH_ZERO_SELF_SIGNED_CERT && GetBoolArg("-allowselfsignedrootcertificates", false))) {
```

Comment:
Hmm... isn't this BoolArg a bit long?
I still struggle with `rpcsslcertificatechainfile`...
IMO: at least certificate could be shorten with cert.

This arg will not get listed anywhere in `--help` or exposed in GUI settings?